signalfx/resource_signalfx_team.go

```diff
@@ -0,0 +1,409 @@
+
+	if err := d.Set("name", t.Name); err != nil {
+		return err
+	}
+	if err := d.Set("description", t.Description); err != nil {
+		return err
+	}
+
+	if len(t.Members) > 0 {
+		members := make([]interface{}, len(t.Members))
+		for i, v := range t.Members {
+			members[i] = v
+		}
+		if err := d.Set("members", schema.NewSet(schema.HashString, members)); err != nil {
+			return err
+		}
+	}
+
+	if len(t.NotificationLists.Critical) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Critical)
+		if err != nil {
+			return err
+		}
+		log.Printf("[DEBUG] SignalFx: CRITICAL %v", nots)
+		d.Set("notifications_critical", nots)
+	}
+	if len(t.NotificationLists.Default) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Default)
+		if err != nil {
+			return err
+		}
+		d.Set("notifications_default", nots)
+	}
+	if len(t.NotificationLists.Info) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Info)
+		if err != nil {
+			return err
+		}
+		d.Set("notifications_info", nots)
+	}
+	if len(t.NotificationLists.Major) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Major)
+		if err != nil {
+			return err
+		}
+		d.Set("notifications_major", nots)
+	}
+	if len(t.NotificationLists.Minor) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Minor)
+		if err != nil {
+			return err
+		}
+		d.Set("notifications_minor", nots)
+	}
+	if len(t.NotificationLists.Warning) > 0 {
+		nots, err := getNotificationsFromAPI(t.NotificationLists.Warning)
+		if err != nil {
+			return err
+		}
+		d.Set("notifications_warning", nots)
+	}
+	return nil
+}
+
+func getNotificationsFromAPI(nots []*notification.Notification) ([]string, error) {
+	results := make([]string, len(nots))
+	for i, not := range nots {
+		s, err := getNotifyStringFromAPI(not)
+		if err != nil {
+			return nil, err
+		}
+		results[i] = s
+	}
+	return results, nil
+}
+
+func teamRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	c, err := config.Client.GetTeam(d.Id())
+	if err != nil {
+		return err
+	}
+
+	return teamAPIToTF(d, c)
+}
+
+func teamUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+	payload, err := getPayloadTeam(d)
+	if err != nil {
+		return err
+	}
+	debugOutput, _ := json.Marshal(payload)
+	log.Printf("[DEBUG] SignalFx: Update Team Payload: %s", string(debugOutput))
+
+	c, err := config.Client.UpdateTeam(d.Id(), payload)
+	if err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] SignalFx: Update Team Response: %v", c)
+
+	d.SetId(c.Id)
+	return teamAPIToTF(d, c)
+}
+
+func teamDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*signalfxConfig)
+
+	return config.Client.DeleteTeam(d.Id())
+}
+
+func teamExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	config := meta.(*signalfxConfig)
+	_, err := config.Client.GetTeam(d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "Bad status 404") {
```

Comment:
Seems a bit hacky.

Reply:
Agreed. I refactored some of this a while back, and this one didn't get it. Will adjust a bit but it's still hacky.